pkg/tsdb/cloudwatch/models/cloudwatch_query.go

```diff
@@ -227,7 +228,7 @@ type metricsDataQuery struct {
 
 // ParseMetricDataQueries decodes the metric data queries json, validates, sets default values and returns an array of CloudWatchQueries.
 // The CloudWatchQuery has a 1 to 1 mapping to a query editor row
-func ParseMetricDataQueries(dataQueries []backend.DataQuery, startTime time.Time, endTime time.Time, dynamicLabelsEnabled,
+func ParseMetricDataQueries(dataQueries []backend.DataQuery, startTime time.Time, endTime time.Time, logger log.Logger, dynamicLabelsEnabled,
```

Comment:
is there a reason for the logger changes?

Reply:
Just think it makes more sense for it to be attached to the instance rather than a specific method.

Reply:
TBH, I'd generally refrain from binding loggers to any kind of program logic.
Loggers should either be centrally managed, or local to a module. Definitely not local to an object or function.
Is this a pattern that's used in other places in the Grafana source code?

Reply:
Here's an example for a package-local logger: https://github.com/grafana/grafana/blob/main/pkg/services/secrets/migrator/migrator.go#L152
But I've also found object-local loggers elsewhere, so there's probably no general guideline on what should be used in Grafana.

Reply:
Yeah I agree with you on this, but for tracing reasons this logger is attached to the [query data execution flow](https://github.com/grafana/grafana/blob/90185bea007086ada4bddb70bc934d10579241c2/pkg/tsdb/cloudwatch/cloudwatch.go#L151), so it cannot be package or module specific. 

Making it object specific is not ideal, but I also don't want it to be passed to a utility function like it would have an affect on the output. Anyway, I think this will become better soon very soon when we've refactored the [query data execution flow](https://github.com/grafana/grafana/issues/57146). Then, we can call the `RequestContextFactoryFunc` from anywhere within the package to retrieve the logger rather than injecting it like this.